### PR TITLE
fix null pointer in PortLock#lock

### PR DESCRIPTION
**Pull request by S11001001**
Part of #6926 .

```
  java.lang.NullPointerException:
  at com.daml.ports.PortLock$Locked.unlock(PortLock.scala:55)
  at com.daml.ports.PortLock$.lock(PortLock.scala:41)
  at com.daml.ports.LockedFreePort$.find(LockedFreePort.scala:15)
  at com.daml.lf.engine.trigger.TriggerServiceFixture$.$anonfun$withTriggerService$1(TriggerServiceFixture.scala:65)
```

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by S11001001** [comment]
> It looks like calling `lock` twice on the same port triggers this reliably.

That case was already tested over here. We need multiple processes for the new path.

https://github.com/digital-asset/daml/blob/7351cf0069822a3d9b0737db780130d05e3a7735/libs-scala/ports/src/test/suite/scala/com/digitalasset/ports/LockedFreePortSpec.scala#L20-L22

File: libs-scala/ports/src/main/scala/com/digitalasset/ports/PortLock.scala [hunk 1]
```diff
@@ -33,14 +33,13 @@ object PortLock {
...
-        Right(locked)
-      } else {
-        locked.unlock()
-        Left(FailedToLock(port))
-      }
+      Option(channel.tryLock())
+        .map(lock => new Locked(port, lock, channel, file))
+        .toRight {
+          channel.close()
+          file.close()
```
[the remark this answers is not in the document]
Comment (leo-da) [reply libs-scala/ports/src/main/scala/com/digitalasset/ports/PortLock.scala:40]: BTW `java.nio.channels.FileChannel#tryLock()` can throw:
```
     * @throws  ClosedChannelException
     * ...
     * @throws  OverlappingFileLockException
     * ...
     * @throws  IOException
     *          If some other I/O error occurs
```
we need to close `channel` on `IOException`.

[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply libs-scala/ports/src/main/scala/com/digitalasset/ports/PortLock.scala:40]: That's a good point too.

**Comment by SamirTalwar-DA** [comment]
Ah, just saw you wrote it can't be tested in a single process. Carry on.

**Comment by cocreature** [comment]
Thanks for the explanation. I agree that a test is not worth it if we need to spin up 2 processes for this.

File: libs-scala/ports/src/main/scala/com/digitalasset/ports/PortLock.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (S11001001) [reply libs-scala/ports/src/main/scala/com/digitalasset/ports/PortLock.scala:40]: Handled in #6926. https://github.com/digital-asset/daml/pull/6926/files#diff-9d9bd06164de62daf31e8b86e3daa175R50-R53

**Pull request merged**